### PR TITLE
Feature weakreference

### PR DESCRIPTION
**Pull request by ErikRBergman**
This is actually a PR that builds on the thread safety PR. There are a few details we should discuss.

### CODE REVIEW

File: src/TinyPubSub/Subscriptions/Subscription.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
+/*
```
Comment (johankson) [inline src/TinyPubSub/Subscriptions/Subscription.cs:1]: Hard to follow what has actually changed when all lines are replaced. Line ending mismatch?

Comment (ErikRBergman) [reply src/TinyPubSub/Subscriptions/Subscription.cs:1]: SubscriptionActions<T> added to handle the weak references, and all subscription files were moved to the /subscriptions/ folder

Comment (johankson) [reply src/TinyPubSub/Subscriptions/Subscription.cs:1]: Ah! Makes sense!

**Comment by johankson** [comment]
What needed discussion?

**Comment by ErikRBergman** [comment]
@johankson 

I wanted you to evaluate the fluent/functional syntax introduced in SubscribeTo.

            TinyPubSubLib.TinyPubSub.SubscribeTo<int>(
                channel,
                s => s
                    .Owner(this)
                    .Tag("ABC123")
                    .WeakReference()
                    .Action(new ReferenceTest<int>(v => testValue = v).DoIt));

**Comment by johankson** [comment]
I'm sorry for dropping the ball on this, do you still feel like including this feature?

**Comment by johankson** [comment]
ping @ErikRBergman

**Comment by johankson** [comment]
I really want to add this feature, could we talk? :D @ErikRBergman

**Comment by ErikRBergman** [comment]
@johankson Sure thing. Was a while ago. :)

**Comment by johankson** [comment]
The worlds slowest PR management. Does it break any existing interfaces? Is it backward compatible? Could you describe how this works in order for me to write the documentation?

**Comment by johankson** [comment]
Also, there are merge conflicts at the moment.

**Comment by johankson** [comment]
Closing this PR, please open it again if you want to merge it.
